pkg/payload/precondition/precondition.go

```diff
@@ -28,10 +28,17 @@ func (e *Error) Cause() error {
 	return e.Nested
 }
 
+// ReleaseContext holds information about the update being considered
+type ReleaseContext struct {
+	// DesiredVersion is the version of the payload being considered.  Remember, this is not guaranteed to be a version
+	// as a customer normally sees it. CI for instance uses invalid versions to ensure you don't rely on this behavior.
```

Comment:
What sort of CI names are you thinking of?  We also claim SemVer in the API, e.g. [here][1].  But putting that asside, I think it makes sense to have the CVO not require SemVer.  How about a docstring like:

```go
// DesiredVersion is the version of the payload being considered.
// While this might be a semantic version, consumers should not
// require SemVer validity so they can handle custom releases
// where the author decided to use a different naming scheme, or
// to leave the version completely unset.
```

or something?
[1]: https://github.com/openshift/api/blob/f263157f58d3b209b6215ad0ea3ef9287316132d/config/v1/types_cluster_version.go#L169-L174

Reply:
The versions in CI parse, but they don't mean anything.  Updating comment.